diff --git a/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+NSNotificationCenter.m b/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+NSNotificationCenter.m
index 67e27db..a770330 100644
--- a/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+NSNotificationCenter.m
+++ b/node_modules/@react-native-firebase/messaging/ios/RNFBMessaging/RNFBMessaging+NSNotificationCenter.m
@@ -77,14 +77,18 @@
   [[RNFBMessagingFIRMessagingDelegate sharedInstance] observe];
 
   RCTRootView *rctRootView;
-  if ([UIApplication sharedApplication].delegate != nil &&
-      [UIApplication sharedApplication].delegate.window != nil &&
-      [UIApplication sharedApplication].delegate.window.rootViewController != nil &&
-      [UIApplication sharedApplication].delegate.window.rootViewController.view != nil &&
-      [[UIApplication sharedApplication].delegate.window.rootViewController.view
-          isKindOfClass:[RCTRootView class]]) {
-    rctRootView =
-        (RCTRootView *)[UIApplication sharedApplication].delegate.window.rootViewController.view;
+  id appDelegate = [UIApplication sharedApplication].delegate;
+
+  if (appDelegate != nil &&
+      [appDelegate respondsToSelector:@selector(window)]) {
+    UIWindow *window = [appDelegate window];
+
+    if (window != nil &&
+        window.rootViewController != nil &&
+        window.rootViewController.view != nil &&
+        [window.rootViewController.view isKindOfClass:[RCTRootView class]]) {
+      rctRootView = (RCTRootView *)window.rootViewController.view;
+    }
   }
 
   // #if !(TARGET_IPHONE_SIMULATOR)
@@ -148,23 +152,29 @@
 }
 
 - (void)application_onDidEnterForeground {
-  if ([UIApplication sharedApplication].delegate != nil &&
-      [UIApplication sharedApplication].delegate.window != nil &&
-      [UIApplication sharedApplication].delegate.window.rootViewController != nil &&
-      [UIApplication sharedApplication].delegate.window.rootViewController.view != nil &&
-      [[UIApplication sharedApplication].delegate.window.rootViewController.view
-          isKindOfClass:[RCTRootView class]]) {
-    RCTRootView *rctRootView =
-        (RCTRootView *)[UIApplication sharedApplication].delegate.window.rootViewController.view;
-
-    if (rctRootView.appProperties != nil &&
-        [rctRootView.appProperties[@"isHeadless"] isEqual:@(YES)]) {
-      NSMutableDictionary *appPropertiesDict = [rctRootView.appProperties mutableCopy];
-      isHeadless = NO;
-      if ([appPropertiesDict objectForKey:@"isHeadless"] != nil &&
-          [appPropertiesDict[@"isHeadless"] isEqual:@([RCTConvert BOOL:@(YES)])]) {
-        appPropertiesDict[@"isHeadless"] = @([RCTConvert BOOL:@(isHeadless)]);
-        rctRootView.appProperties = appPropertiesDict;
+  id appDelegate = [UIApplication sharedApplication].delegate;
+
+  if (appDelegate != nil &&
+      [appDelegate respondsToSelector:@selector(window)]) {
+    UIWindow *window = [appDelegate window];
+
+    if (window != nil &&
+        window.rootViewController != nil &&
+        window.rootViewController.view != nil &&
+        [window.rootViewController.view isKindOfClass:[RCTRootView class]]) {
+      RCTRootView *rctRootView =
+          (RCTRootView *)window.rootViewController.view;
+
+      if (rctRootView.appProperties != nil &&
+          [rctRootView.appProperties[@"isHeadless"] isEqual:@(YES)]) {
+        NSMutableDictionary *appPropertiesDict = [rctRootView.appProperties mutableCopy];
+        isHeadless = NO;
+
+        if ([appPropertiesDict objectForKey:@"isHeadless"] != nil &&
+            [appPropertiesDict[@"isHeadless"] isEqual:@([RCTConvert BOOL:@(YES)])]) {
+          appPropertiesDict[@"isHeadless"] = @([RCTConvert BOOL:@(isHeadless)]);
+          rctRootView.appProperties = appPropertiesDict;
+        }
       }
     }
   }
